### DISCUSSION TITLE
项目无法运行

### DISCUSSION THREAD

**Discussion by ghost** [shadowsocks/shadowsocks-android#2742] [Q&A]
项目无法运行

执行如下命令：
cd core/src/main/rust/shadowsocks-rust
rustup target add armv7-linux-androideabi aarch64-linux-android i686-linux-android x86_64-linux-android
cargo build
build项目

错误如下：
> Task :core:cargoBuildArm FAILED
Execution failed for task ':core:cargoBuildArm'.
> A problem occurred starting process 'command 'rustc''

**Comment by dev4u** [comment]
建议检查一下你的rust编译环境

**Reply by Izzq** [reply]
我也遇到同样问题，请问你在AndroidStudio能运行起来吗？

**Comment by ghost** [comment]
rustup --version
>rustup 1.23.1 (3df2264a9 2020-11-30)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.49.0 (e1884a8e3 2020-12-29)`

cat .bash_profile
>source "$HOME/.cargo/env"

cat $HOME/.cargo/env
```shell
#!/bin/sh
# rustup shell setup
# affix colons on either side of $PATH to simplify matching
case ":${PATH}:" in
    *:"$HOME/.cargo/bin":*)
        ;;
    *)
        # Prepending path in case a system-installed rustc needs to be overridden
        export PATH="$HOME/.cargo/bin:$PATH"
        ;;
esac
```

运行步骤：
`cd core/src/main/rust/shadowsocks-rust`
`rustup target add armv7-linux-androideabi aarch64-linux-android i686-linux-android x86_64-linux-android`
>info: component 'rust-std' for target 'armv7-linux-androideabi' is up to date
info: component 'rust-std' for target 'aarch64-linux-android' is up to date
info: component 'rust-std' for target 'i686-linux-android' is up to date
info: component 'rust-std' for target 'x86_64-linux-android' is up to date

`cargo build`
>Finished dev [unoptimized + debuginfo] target(s) in 0.38s

**Reply by MarinaTsang** [reply]
请问解决了吗？求同步答案

**Comment by dev4u** [comment]
兄弟，编译环境不单是这些，还有你编译的依赖是不是正常加载、feather、是执行到什么状态出问题、会不会是oom……

**Comment by myflyingfish** [comment]
好了吗 我的同样的问题

**Reply by MarinaTsang** [reply]
请问问题解决了吗

**Reply by GeneralZMX** [reply]
大佬们怎么解决的

**Reply by winnxiegang** [reply]
大佬们怎么解决的 求解

**Comment by GeneralZMX** [comment]
大佬们怎么解决的

**Comment by winnxiegang** [comment]
大佬们怎么解决的 求解决方法
